test/decoder_test.exs

```diff
@@ -20,15 +20,16 @@ defmodule MozartFetcher.DecoderTest do
       assert Decoder.decode_envelope(json, %Envelope{}) == expected
     end
 
-    test "when the JSON contains invalid keys we rescue and return an error" do
-      json = ~s({"head":[],"bodyInLine":"<DIV id=\\"site-container\\">","bodyLast":[]})
+    test "when the %Envelope{} has invalid keys we still return an empty %Envelope" do
+      json = ~s({"head":[],"foobar":"<DIV id=\\"site-container\\">","bodyLast":[]})
 
-      assert Decoder.decode_envelope(json, %Envelope{}) == {:error}
+      assert Decoder.decode_envelope(json, %Envelope{}) ==
+               {:ok, %MozartFetcher.Envelope{bodyInline: "", bodyLast: [], head: []}}
     end
```

Comment:
In the statement:  `%Envelope{}` (struct) has invalid keys, but in test it's referring to the incoming JSON body?

Reply:
in the json variable the key `bodyLast` is `foobar` ill try and make it a bit clearer though if you think that is necessary, maybe something like:
`when the json does not conform to the %Envelope{} structs keys we return an error`?

the return value of this is going to change back to {:error} instead of an empty %Envelope{} as the envelope class will return the empty %Envelope{}

Reply:
`test "when the JSON contains invalid keys for an %Envelope{} we return an error" do`
Gonna go with this thinks its clearer

Reply:
Note the test and code logic does not return an error. It returns an empty struct.

Reply:
It makes sense to use "the JSON" since you already used that in the preceding case.

Reply:
Sorry I think I was a bit confusing in my message, the behaviour has been updated in my latest commits and the wording has been made clearer now. Thanks :)